src/MusicTheory/PitchClass/Enharmonic.elm

```diff
@@ -0,0 +1,245 @@
+module MusicTheory.PitchClass.Enharmonic exposing
+    ( atMostOneAccidental
+    , exactSpelling
+    , naturalOrSingleFlat
+    , naturalOrSingleSharp
+    , simplified
+    , simplifiedToString
+    , toString
+    )
+
+import MusicTheory.Accidental exposing (Accidental(..))
+import MusicTheory.Internals.PitchClass as Internal
+import MusicTheory.Letter exposing (Letter(..))
+import MusicTheory.PitchClass as PitchClass exposing (PitchClass)
+
+
+{-| Returns the letter and accidental of a pitch class if the letter is raised or lowered such that it can be expressed in terms of a valid accidental.
+
+    (PitchClass.pitchClass F Sharp |> exactSpelling) == Just ( F, Sharp )
+
+-}
+exactSpelling : PitchClass -> Maybe ( Letter, Accidental )
```

Comment:
Again, I disagree with the idea of two representations of a `PitchClass`. Shouldn't this return a `PitchClass`?

Reply:
I don't see it as another representation for pitch class. I see it as the result of different accessors to determine a concrete spelling.

Reply:
Pitch class is an opaque thing that you cannot really spell out. Returning offset in terms of int is a no go because it supports states that are not defined in music theory. Therefore the type Accidental is absolutely necessary to represent the result of an transformation to something that is valid in theory. `Maybe int` is a wrong representation IMO.